safekeeper/src/wal_backup.rs

```diff
@@ -408,10 +409,11 @@ pub async fn backup_lsn_range(
 
 async fn backup_single_segment(
     seg: &Segment,
-    timeline_dir: &Path,
-    workspace_dir: &Path,
+    timeline_dir: &Utf8Path,
+    workspace_dir: &Utf8Path,
 ) -> Result<Segment> {
-    let segment_file_path = seg.file_path(timeline_dir)?;
+    let segment_file_path = Utf8PathBuf::from_path_buf(seg.file_path(timeline_dir.as_std_path())?)
+        .expect("non-Unicode path");
```

Comment:
If `Segment::file_path` operated on Utf8Path then this expect would be statically avoided.

Reply:
Hmm, seems like I missed this part. :cold_sweat: